Code/Framework/AzQtComponents/AzQtComponents/Components/Widgets/Internal/Rectangle.h

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#pragma once
+
+#if !defined(Q_MOC_RUN)
+#include <AzQtComponents/AzQtComponentsAPI.h>
+
+#include <QWidget>
+#endif
+
+class QPaintEvent;
+
+namespace AzQtComponents::Internal
+{
+    //! A rectangle of a single color
+    //! To be used when filling some space with a solid color
+    //! is not feasible through styling.
+    class AZ_QT_COMPONENTS_API Rectangle : public QWidget
```

Comment:
suggestion/thought: Not 100% sure and maybe this doesn't matter as we're inside the `AzQtComponents::Internal` namespace but `Rectangle` does seem like too general a name (as it infers a more mathematical type). Maybe `FilledRectangle` or `RectangleWidget` might be better 🤔 Not a huge deal but just might make reading/understanding the code a bit easier in future.

Reply:
`RectangleWidget` sounds nice. I'll do the change.

Reply:
Awesome thanks!